File: 251004/Kalitsko/3lab/discussion/src/main/java/com/example/rest/mapper/PostMapper.java
package com.example.rest.mapper;

import com.example.rest.dto.PostRequestTo;
import com.example.rest.dto.PostResponseTo;
import com.example.rest.entity.Post;
import org.mapstruct.Mapper;
import org.mapstruct.Mapping;

@Mapper(componentModel = "spring")
public interface PostMapper {
    Post toEntity(PostRequestTo request);
    PostResponseTo toResponse(Post post);
}

File: 251003/Pankratiev/rest/discussion/src/main/kotlin/bsuir/dc/discussion/mapper/PostMapper.kt
package bsuir.dc.discussion.mapper

import bsuir.dc.discussion.dto.from.PostRequestTo
import bsuir.dc.discussion.dto.to.PostResponseTo
import bsuir.dc.discussion.entity.Post
import bsuir.dc.discussion.entity.PostKey

fun PostRequestTo.toEntity(country: String): Post {
    return Post(
        PostKey(country = country, id = this.id),
        issueId = this.issueId,
        content = this.content,
    )
}

fun Post.toResponse(): PostResponseTo {
    return PostResponseTo(
        id = this.key.id,
        issueId = this.issueId,
        content = this.content
    )
}

